packages/belos/src/BelosBlockGmresIter.hpp

```diff
@@ -852,7 +859,7 @@ class BlockGmresIter : virtual public GmresIteration<ScalarType,MV,OP> {
           sigma = blas.DOT( blockSize_, &(*R_)(curDim+j+1,curDim+j),
                             1, &(*z_)(curDim+j+1,i), 1);
           sigma += (*z_)(curDim+j,i);
-          sigma *= beta[curDim+j];
+          sigma *= SCT::conjugate(beta[curDim+j]);
           blas.AXPY(blockSize_, ScalarType(-sigma), &(*R_)(curDim+j+1,curDim+j),
                     1, &(*z_)(curDim+j+1,i), 1);
           (*z_)(curDim+j,i) -= sigma;
```

Comment:
I was wondering why `-sigma` needs to be type-casted to ScalarTyppe.

Reply:
@iyamazaki Belos has limited the use of LAPACK across the solvers due to the lack of templated implementations in Teuchos::LAPACK.  For users who want to use ScalarTypes other than the standard float,double,complex<float>, complex<double>, this can limit the number of available iterative solvers.  So, yes, it is just more flexible this way.

Reply:
@iyamazaki I believe the -sigma type-cast was required by a reported compiler warning.

Reply:
Ah, that makes sense. Thank you!!